src/extensions/default/CodeFolding/Prefs.js

```diff
@@ -0,0 +1,127 @@
+                           {name: strings.ENABLE_REGION_FOLDING, description: strings.ENABLE_REGION_FOLDING});
+    prefs.definePreference("fadeFoldButtons", "boolean", false,
+                           {name: strings.FADE_FOLD_BUTTONS, description: strings.FADE_FOLD_BUTTONS_HELP});
+    prefs.definePreference("maxFoldLevel", "number", 2,
+                           {name: strings.MAX_FOLD_LEVEL, description: strings.MAX_FOLD_LEVEL_HELP});
+    prefs.definePreference("folds", "object", {});
+
+    /**
+        Simplifies the fold ranges into an array of pairs of numbers.
+        @param {!{number: {from: {ch, line}, to: {ch, line}} folds the raw fold ranges indexed by line numbers
+        @returns {number: Array<Array<number>>} an object whose keys are line numbers and the values are array
+        of two 2-element arrays. First array contains [from.line, from.ch] and the second contains [to.line, to.ch]
+    */
+    function simplify(folds) {
+        if (!folds) {
+            return;
+        }
+        var res = {}, range;
+        Object.keys(folds).forEach(function (line) {
+            range = folds[line];
+            res[line] = Array.isArray(range) ? range : [[range.from.line, range.from.ch], [range.to.line, range.to.ch]];
+        });
+        return res;
+    }
+    /**
+        Inflates the fold ranges stored as simplified numeric arrays. The inflation converts the data into
+        objects whose keys are line numbers and whose values are objects in the format {from: {line, ch}, to: {line, ch}}.
+        @param {number: Array<Array<number>>}  folds the simplified fold ranges
+        @returns {number: {from: {line, ch}, to: {line, ch}}}
+    */
+    function inflate(folds) {
+        if (!folds) {
+            return;
+        }
+         //transform the folds into objects with from and to properties
+        var ranges = {}, obj;
+        Object.keys(folds).forEach(function (line) {
+            obj = folds[line];
+            ranges[line] = {from: {line: obj[0][0], ch: obj[0][1]}, to: {line: obj[1][0], ch: obj[1][1]}};
+        });
+
+        return ranges;
+    }
+
+    /**
+        Gets the line folds saved for the specified path.
+        @param {string} path the document path
+        @returns {number: {from: {line, ch}, to: {line, ch}}} the line folds for the document at the specified path
+    */
+    function get(path) {
+        store = (prefs.get(foldsKey) || {});
+        return inflate(store[path]);
+    }
+
+    /**
+        Saves the line folds specified
+    */
+    function set(path, folds) {
+        store[path] = simplify(folds);
+        prefs.set(foldsKey, store);
+    }
+
+    /**
+        Get the code folding setting with the specified key from the store
+        @param {!string} key The key for the setting to retrieve
+        @returns {string} the setting with the specified key
+    */
+    function getSetting(key) {
+        return prefs.get(key, PreferencesManager.CURRENT_PROJECT);
```

Comment:
@MiguelCastillo thanks, that was how I thought it worked. All I am trying to do is to override any code-folding preferences set at the user or default level with one set at the project level. For instance placing `    "code-folding.enabled": false
` in .brackets.json file to disable code-folding. I couldn't get this to work consistently yesterday so I thought I'd 'force' it to look at the project scope using the second parameter [as suggested by the wiki](https://github.com/adobe/brackets/wiki/Preferences-System#looking-up-prefs). At the moment, when I put a breakpoint in developer console to debug, it works as expected. Without the breakpoint it doesn't seem to be reading the correct preference value.
Any insights or questions would be useful. I'll also have a closer look later today.
Cheers.